packages/autoloader/src/class-plugins-handler.php

```diff
@@ -2,128 +2,155 @@
+		$this->plugin_locator = $plugin_locator;
+		$this->cache_handler  = $cache_handler;
 	}
 
 	/**
-	 * Ensure the plugin has its own directory and not a single-file plugin.
-	 *
-	 * @param string $plugin Plugin name, may be prefixed with "/".
+	 * Gets the directory of the current plugin.
 	 *
-	 * @return bool
+	 * @return string
 	 */
-	public function is_directory_plugin( $plugin ) {
-		return strlen( $plugin ) > 1 && false !== strpos( $plugin, '/', 1 );
+	public function get_current_plugin() {
+		// Escape from `vendor/__DIR__` to plugin directory.
+		return dirname( dirname( __DIR__ ) );
 	}
 
 	/**
-	 * Returns an array containing the names of plugins that are activating via a request.
+	 * Gets all of the active plugins we can find.
 	 *
-	 * @return array An array of names of the activating plugins or an empty array.
+	 * @return string[]
 	 */
-	private function get_plugins_activating_via_request() {
-
-		// phpcs:disable WordPress.Security.NonceVerification.Recommended
-		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.MissingUnslash
-		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
-
-		$action = isset( $_REQUEST['action'] ) ? $_REQUEST['action'] : false;
-		$plugin = isset( $_REQUEST['plugin'] ) ? $_REQUEST['plugin'] : false;
-		$nonce  = isset( $_REQUEST['_wpnonce'] ) ? $_REQUEST['_wpnonce'] : false;
-
-		/**
-		 * Note: we're not actually checking the nonce here becase it's too early
-		 * in the execution. The pluggable functions are not yet loaded to give
-		 * plugins a chance to plug their versions. Therefore we're doing the bare
-		 * minimum: checking whether the nonce exists and it's in the right place.
-		 * The request will fail later if the nonce doesn't pass the check.
-		 */
-
-		// In case of a single plugin activation there will be a plugin slug.
-		if ( 'activate' === $action && ! empty( $nonce ) ) {
-			return array( $this->create_plugin_path( wp_unslash( $plugin ) ) );
+	public function get_active_plugins() {
+		global $jetpack_autoloader_activating_plugins_paths;
+		global $jetpack_autoloader_including_latest;
+
+		// We're going to build a unique list of plugins from a few different sources
+		// to find all of our "active" plugins. While we need to return an integer
+		// array, we're going to use an associative array internally to reduce
+		// the amount of time that we're going to spend checking uniqueness
+		// and merging different arrays together to form the output.
+		$active_plugins = array();
+
+		// Make sure that plugins which have activated this request are considered as "active" even though
+		// they probably won't be present in any option.
+		if ( is_array( $jetpack_autoloader_activating_plugins_paths ) ) {
+			foreach ( $jetpack_autoloader_activating_plugins_paths as $path ) {
+				$active_plugins[ $path ] = $path;
+			}
+		}
+
+		// This option contains all of the plugins that have been activated via the interface.
```

Comment:
What does "activated via the interface" mean? I think the `activate_plugins` option contains all of the currently active plugins, regardless of how they were activated.

Reply:
Who knows :) I've adjusted the comment!